sources/web/datalab/polymer/modules/drive-file-manager/drive-file-manager.ts

```diff
@@ -116,27 +126,33 @@ class DriveFileManager extends BaseFileManager {
       // TODO - create the real path to this object, or figure out
       // a better way to handle not having the full path in the breadcrumbs
       const fileId = path;  // We assume the entire path is one fileId
-      const datalabFile: DriveFile = new DriveFile({
-        id: new DatalabFileId(fileId, FileManagerType.DRIVE),
-      } as DatalabFile);
+      const datalabFile: DriveFile = new DriveFile(
+        new DatalabFileId(fileId, FileManagerType.DRIVE),
+        path,
+        DatalabFileType.DIRECTORY,
+      );
       return [datalabFile];
     }
   }
 
   protected _fromUpstreamFile(file: gapi.client.drive.File) {
-    const datalabFile: DriveFile = new DriveFile({
-      icon: file.mimeType === DriveFileManager._directoryMimeType ?
-                              file.iconLink : 'editor:insert-drive-file',
-      id: new DatalabFileId(file.id, FileManagerType.DRIVE),
-      name: file.name,
-      type: file.mimeType === DriveFileManager._directoryMimeType ?
+    const driveFile = new DriveFile(
+      new DatalabFileId(file.id, FileManagerType.DRIVE),
+      file.name,
+      file.mimeType === DriveFileManager._directoryMimeType ?
                               DatalabFileType.DIRECTORY :
                               DatalabFileType.FILE,
-    } as DatalabFile);
-    if (datalabFile.name.endsWith('.ipynb')) {
-      datalabFile.type = DatalabFileType.NOTEBOOK;
+      file.mimeType === DriveFileManager._directoryMimeType ?
+                              file.iconLink : 'editor:insert-drive-file',
+    );
+    if (driveFile.type === DatalabFileType.FILE && driveFile.name.endsWith('.ipynb')) {
+      driveFile.type = DatalabFileType.NOTEBOOK;
+    }
+    driveFile.lastModified = new Date(file.modifiedTime).toDateString();
```

Comment:
Converting this date to a string here will make it difficult to sort on that column. Maybe you could do the string conversion in DatalabFile.getColumnValues instead. Or return a value here that has a toString() that generates the right display value, but that also can be sorted.

Reply:
Good point.
I think we should do this more properly by defining the column type along with its name in the item-list component, so that ordering can use a custom sort function. I'd like to do that as a separate change.

Reply:
I'd ask you to add a TODO if it didn't require another round-trip for approval. Or maybe you are planning on working on that right away anyway.

Reply:
Yup, it's next on my list. :)